webboot/webboot.go

```diff
@@ -23,9 +40,88 @@ var (
 	ipv6    = flag.Bool("ipv6", true, "use IPV6")
 )
 
+func wgetArgs(argURL string) (read io.ReadCloser, outFile string, werr error) {
+	//checks if argURL is a key in osSystems map
+	if _, ok := osSystems[argURL]; !ok {
+		//if argURL is not the map, use it as a URL link for wget
+		read, werr = wget(argURL)
+	} else {
+		// if argURL is in osSystems map, use its download link for wget
+		read, werr = wget(osSystems[argURL].DownloadLink)
+		outFile = argURL
+	}
+	return read, outFile, werr
```

Comment:
in one of the ifs you don't set outFile (and it's not a file anyway, it's a name). You should always set it to something. It makes following code simpler. 

change the name to URL, since that's what it is
or change it to url, if you don't like caps, but URL stands for
Universal Resource Locator, and Go convention is to make acronyms all caps.

not sure you need the comments on lines 46 and 59, the code is really clear
also consider you use osSsystems[argURL] in two places; why not this?
(don't name things argTHIS and argTHAT, we know it's an argument by looking at code)
(don't name return types unless you really need to, and in this case you don't)

This function also kinda complicated with duplicated code. you could do this:

func getURL(URL string) ( io.ReadCloser,  string, error) {
if u, ok := osSystems[URL]; ok {
URL = u.DownloadLink
}
		read, err \:= wget(URL)
                 return read, URL, err
}

much less code.

Reply:
Yes, I notice what you were saying. I checked the second half of that implementation somewhere else, but I paired the two together. Now, I either keep the filename as the arg the user passed in such as "tinycore" or I set it the base of the url if the user provided me a URL link. I have gotten rid of argTHIS and argTHAT.

Reply:
I have tweaked the code slightly due to other code maintenance while also recognizing this simplicity and using it.